axiom/nr_segment_datastore.c

```diff
@@ -12,6 +12,36 @@
 #include "util_sql.h"
 #include "util_logging.h"
 
+static bool create_instance_metric(nr_segment_t* segment,
+                                   const char* product,
+                                   nr_segment_datastore_t* datastore,
+                                   const nr_datastore_instance_t* instance) {
+  /*
+   * If a datastore instance was provided, we need to add the relevant data to
+   * the segment and the relavant metrics.
+   */
+  nrtxn_t* txn = segment->txn;
+  char* instance_metric = NULL;
+  if (NULL == instance || 0 == txn->options.instance_reporting_enabled) {
```

Comment:
Check for `txn` being `NULL`?

Reply:
`txn` is checked for `NULL` by the caller. `datastore` is a local variable of the caller.